stix2/registration.py

```diff
@@ -146,30 +155,31 @@ def _register_extension(
     """
     ext_type = new_extension._type
 
-    # Need to check both toplevel and nested properties
-    prop_groups = [new_extension._properties]
-    if hasattr(new_extension, "_toplevel_properties"):
-        prop_groups.append(new_extension._toplevel_properties)
-    prop_names = itertools.chain.from_iterable(prop_groups)
-
     _validate_type(ext_type, version)
-
-    if not new_extension._properties:
-        raise ValueError(
-            "Invalid extension: must define at least one property: " +
-            ext_type,
-        )
-
     if version == "2.1":
         if not (ext_type.endswith('-ext') or ext_type.startswith('extension-definition--')):
             raise ValueError(
                 "Invalid extension type name '%s': must end with '-ext' or start with 'extension-definition--<UUID>'." %
                 ext_type,
             )
 
-        for prop_name in prop_names:
-            if not re.match(PREFIX_21_REGEX, prop_name):
-                raise ValueError("Property name '%s' must begin with an alpha character." % prop_name)
+    tl_props = getattr(new_extension, "_toplevel_properties")
```

Comment:
This and the code below would be more elegant if we changed `custom.py` to set the `_CustomExtension._toplevel_properties` attribute to `None` rather than not defining it at all for non-top-level-property extensions.

Reply:
So instead you would have:

```python
tl_props = new_extension._toplevel_properties
```

And the code below would not change?  The change doesn't seem all that significant.  What did you have in mind?

Reply:
Yep. It'd mostly just mean that all extensions have a `_top_level_properties` rather than having to do `getattr()`s on it. In my head, being able to always access an expected attribute and react when it's set to a value defining "not present on the represented object" feels nice. I'm not stressed about changing this if you're satisfied with the code as it stands though